src/Mod/Sketcher/Gui/Utils.cpp

```diff
@@ -328,45 +328,36 @@ double SketcherGui::GetPointAngle(const Base::Vector2d& p1, const Base::Vector2d
 }
 
 // Set the two points on circles at minimal distance
-// in concentric case return false and set points on relative X axis
-bool SketcherGui::GetCirclesMinimalDistance(const Part::GeomCircle *circle1, const Part::GeomCircle *circle2, Base::Vector3d *point1, Base::Vector3d *point2)
+// in concentric case set points on relative X axis
+void SketcherGui::GetCirclesMinimalDistance(const Part::GeomCircle *circle1, const Part::GeomCircle *circle2, Base::Vector3d &point1, Base::Vector3d &point2)
 {
-    auto center1 = circle1->getCenter();
-    auto center2 = circle2->getCenter();
     double radius1 = circle1->getRadius();
     double radius2 = circle2->getRadius();
-    Base::Vector3d v = center2 - center1;
+
+    point1 = circle1->getCenter();
+    point2 = circle2->getCenter();
+
+    Base::Vector3d v = point2 - point1;
     double length = v.Length();
-    auto p1 = Base::Vector3d(0., 0., 0.);
-    auto p2 = Base::Vector3d(0., 0., 0.);
-    bool retval = false;
 
-    v = v.Normalize();
     if (length == 0) { //concentric case
-        p1 = center1 + radius1 * Base::Vector3d(1.,0.,0.);
-        p2 = center1 + radius2 * Base::Vector3d(1.,0.,0.);
+        point1.x += radius1;
+        point2.x += radius2;
     } else {
+        v = v.Normalize();
         if (length <= std::max(radius1, radius2)){ //inner case
             if (radius1 > radius2){
-                p1 = center1 + v * radius1;
-                p2 = center2 + v * radius2;
+                point1 += v * radius1;
+                point2 += v * radius2;
             } else {
-                p1 = center1 - v * radius1;
-                p2 = center2 - v * radius2;
+                point1 += -v * radius1;
+                point2 += -v * radius2;
```

Comment:
There is operator -=, but actually the way it is here is more visual, as one can spot the differences between calls.

BTW, I like the refactoring.